processor/src/main/java/org/mapstruct/ap/internal/model/Mapper.java

```diff
@@ -155,7 +155,11 @@ private Mapper(TypeFactory typeFactory, String packageName, String name,
         this.customPackage = customPackage;
         this.customImplName = customImplName;
         customAnnotations.forEach( this::addAnnotation );
-
+        TypeElement typeElement = this.getMapperDefinitionType().getTypeElement();
+        if ( typeElement.getAnnotation( Deprecated.class ) != null
+                && typeFactory.isTypeAvailable( Deprecated.class.getCanonicalName() )) {
```

Comment:
I don't think that we need to check if `java.lang.Deprecated` is available. This comes from `java.base`, which means it will always be available on the classpath.

Reply:
Thank you for pointing it out, it has been changed